app/services/url_parser.py

```diff
@@ -0,0 +1,21 @@
+from typing import Literal, Tuple
+from urllib.parse import urlparse
+
+from fastapi import HTTPException
+
+
+def parse_twitter_url(url: str) -> Tuple[Literal["user", "tweet"], str]:
+    parsed = urlparse(url)
+
+    if parsed.netloc != "twitter.com":
+        raise HTTPException(400, "'url' argument is invalid!")
```

Comment:
Wait, does this mean that calling `check` endpoint with `username` is invalid now? That' not good